src/foundry/client/core/socket.d.mts

```diff
@@ -11,6 +11,51 @@ declare global {
     static dispatch<DatabaseAction extends keyof DatabaseOperationMap>(
       eventName: string,
       request: DocumentSocketRequest<DatabaseAction> | ManageCompendiumRequest,
-    ): Promise<SocketResponse>;
+    ): Promise<SocketInterface.SocketResponse>;
+  }
+
+  namespace SocketInterface {
+    /** @remarks Copied from `resources/app/common/types.mjs` */
+    type RequestData = Record<string, unknown> | Record<string, unknown>[] | string | string[];
+
+    /** @remarks Copied from `resources/app/common/types.mjs` */
+    interface SocketRequest {
+      /** Additional options applied to the request */
+      options?: Record<string, unknown> | undefined;
```

Comment:
I don't know if it's worth trying to type any more strongly because it'd require typing everything on the wire but I suppose `AnyObject` is still probably a good idea here.